Enterprise-CMCS/eAPD
# Tforkner/4196 sharepoint endpoints

Resolves #4196

### Description
Creates two endpoints to be used by Kong

### Automated test cases written
Created standard endpoint and unit test.

| Given | When | Then | Type (jest, tap, cypress) |
| ----- | ---- | ---- | ------------------------- |
| submitted APDs | GET all submitted APDs | returns all APDs with submitted status | tap |
| status updates for APDs | the updates are empty | it handles that gracefully | tap |
| status updates for APDs | there is one valid update | the APD status is updated | tap |
| status updates for APDs | there are multiple updates | the APD statuses are updated | tap |
| status updates for APDs | there are valid and invalid statuses | it handles it gracefully, the valid status is updated and the invalid isn't | tap |
| a GET call to /apds/submissions | the user IP doesn't match | a 403 is returned | jest |
| a GET call to /apds/submissions | the user IP does match | the APDs and a status 200 is returned | jest |
| a PATCH call to /apds/submissions | the user IP doesn't match | a 403 is returned | tap |
| a PATCH call to /apds/submissions | the user IP does match | a 403 is returned | tap |
| a PATCH call to /apds/submissions | the user IP does match but the body is invalid | a 400 is returned | tap |
| a PATCH call to /apds/submissions | the user IP does match and the body is valid | a 204 is returned | tap |
| a PATCH call to /apds/submissions | the user IP doesn't match | a 403 is returned | jest |
| a PATCH call to /apds/submissions | the user IP does match but the body is invalid | a 400 is returned | jest |
| a PATCH call to /apds/submissions | the user IP does match and the body is valid | a 304 is returned | jest |


### Steps to manually verify this change

1. using Postman, GET /apds/submissions and verify that you get a 403
2. using Postman, PATCH /apds/submissions and verify that you get a 403
3. add you public IP to the flag in LaunchDarkly
4. using Postman, GET /apds/submissions and verify that you get a 200
5. using Postman, PATCH /apds/submissions with no body and verify that you get a 400
6. using Postman, PATCH /apds/submissions with a raw json body and verify that you get a 204, the valid apdId should be replaced by an actual apdId which you can find by opening any apd in the browser and copying the long alphanumberic string after apd/ in the url
```    
    [{
        "apdId": "valid apdId",
        "newStatus": "approved"
    }]
```

### This pull request is ready to code review when

      tests, refactor

### This pull request is ready to test when


### This pull request is ready to review when the QA has


### This pull request can be merged when
